core/src/main/scala/org/apache/spark/resource/ResourceAllocator.scala

```diff
@@ -38,11 +37,13 @@ private[spark] trait ResourceAllocator {
    * For task resources ([[org.apache.spark.scheduler.ExecutorResourceInfo]]), this value
    * can be a multiple, such that each address can be allocated up to [[slotsPerAddress]]
    * times.
-   *
-   * TODO Use [[org.apache.spark.util.collection.OpenHashMap]] instead to gain better performance.
```

Comment:
cc @Ngone51 @tgravescs FYI

The TODO is add by @Ngone51  and @tgravescs make `addressAvailabilityMap` lazy